### DISCUSSION TITLE
Why completion popup takes so long to appear?

### DISCUSSION THREAD

**Discussion by nowaylifer** [LazyVim/LazyVim#2982] [Q&A] [answered]
https://github.com/LazyVim/LazyVim/assets/116919411/48abecbb-8ce0-4155-af79-b491e459d8d6

**Comment by amaanq** [comment]
LuaLS is somewhat slow in general, so there might not be much you can do

**Comment by dpetka2001** [comment, marked as answer]
It's the plugins indexing by `lua_ls`. 

You are able to disable this by setting in your `.neoconf.json` `neodev.library.plugins: false`. 

Another way you might want to explore is disabling `neodev` as a whole and set up `lua_ls` as per the [nvim-lspconfig docs](https://github.com/neovim/nvim-lspconfig/blob/b73f30d113ace671bcf1f731dfc66987880410e6/lua/lspconfig/server_configurations/lua_ls.lua#L47-L72) on your own and provide a `.luarc.json` file with only the plugins that are of interest to you, in order to decrease the amount of indexing `lua_ls` will do and still be able to have a decent response in the auto-completion popup.
